File: packages/mc-prism/packages/tools/fontsource_fetcher.py
#!/usr/bin/env python3

import requests
import json
from typing import Dict

class FontSourceFetcher:
    BASE_URL = "https://api.fontsource.org"

    def get_font_list(self) -> Dict[str, str]:
        """
        Fetch the list of available fonts from Fontsource API
        Returns: Dictionary with font_id as key and source as value
        """
        response = requests.get(f"{self.BASE_URL}/fontlist")
        if response.status_code != 200:
            raise Exception(f"Failed to fetch font list: {response.status_code}")

        return json.loads(response.text)

# Only run this if the file is run directly
if __name__ == "__main__":
    fetcher = FontSourceFetcher()
    fonts = fetcher.get_font_list()
    print("API Response Structure:")
    print(type(fonts))
    print("\nFirst few entries:")
    for i, (key, value) in enumerate(fonts.items()):
        if i >= 3:
            break
        print(f"\nKey: {key}")
        print(f"Value: {value}")

File: packages/mc-prism/packages/tools/fontsource_fetcher.js
const https = require('https');

class FontSourceFetcher {
  constructor() {
    this.BASE_URL = 'https://api.fontsource.org';
  }

  getFontList() {
    return new Promise((resolve, reject) => {
      https
        .get(`${this.BASE_URL}/fontlist`, (response) => {
          let data = '';

          response.on('data', (chunk) => {
            data += chunk;
          });

          response.on('end', () => {
            if (response.statusCode === 200) {
              resolve(JSON.parse(data));
            } else {
              reject(
                new Error(`Failed to fetch font list: ${response.statusCode}`)
              );
            }
          });
        })
        .on('error', (error) => {
          reject(error);
        });
    });
  }
}

// Only run this if the file is run directly
if (require.main === module) {
  (async () => {
    const fetcher = new FontSourceFetcher();
    try {
      const fonts = await fetcher.getFontList();
      console.log('API Response Structure:');
      console.log(typeof fonts);
      console.log('\nFirst few entries:');
      Object.entries(fonts)
        .slice(0, 3)
        .forEach(([key, value]) => {
          console.log(`\nKey: ${key}`);
          console.log(`Value: ${value}`);
        });
    } catch (error) {
      console.error(error.message);
    }
  })();
}
